lib/nats.js

```diff
@@ -615,7 +615,7 @@ Client.prototype.createConnection = function() {
     // otherwise in addition to the leak events will fire fire
```

Comment:
fire fire

Reply:
From looking at it, the stream cleanup code from:

```Client.prototype.closeStream = function() {
  if (this.stream !== null) {
    this.stream.end();
    this.stream.destroy();
```
was duplicated to the places were we could fail to close the stream.
 ` this.stream.removeAllListeners();` is the fix for #45 multiple notifications when listeners were left out.